### PR TITLE
NonlinearLeastSquaresProblem Solvers: Levenberg & Gauss Newton

### PR DESCRIPTION
**Pull request by avik-pal**
- [x] Wait on https://github.com/SciML/SciMLBase.jl/pull/500
- [x] Wait on https://github.com/SciML/DiffEqBase.jl/pull/933
- [x] Levenberg
- [x] Gauss Newton
- [x] Tests
- [x] Look into how convergence checks are implemented in other libraries

### CODE REVIEW

File: docs/src/solvers/NonlinearLeastSquaresSolvers.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+`LevenbergMarquardt` is a good choice for most problems.
+
+## Full List of Methods
+
+  - `LevenbergMarquardt()`: An advanced Levenberg-Marquardt implementation with the
+    improvements suggested in the [paper](https://arxiv.org/abs/1201.5885) "Improvements to
+    the Levenberg-Marquardt algorithm for nonlinear least-squares minimization". Designed for
+    large-scale and numerically-difficult nonlinear systems.
+  - `GaussNewton()`: An advanced GaussNewton implementation with support for efficient
```
[the remark this answers is not in the document]
Comment (avik-pal) [reply docs/src/solvers/NonlinearLeastSquaresSolvers.md:18]: Yeah

[the remark this answers is not in the document]
Comment (ChrisRackauckas) [reply docs/src/solvers/NonlinearLeastSquaresSolvers.md:18]: Can we document that as an available method and add a test there as well?

[the remark this answers is not in the document]
Comment (avik-pal) [reply docs/src/solvers/NonlinearLeastSquaresSolvers.md:18]: Done. I have also added a warning about the robustness of that!

**Comment by avik-pal** [comment]
This is ready for merge!

**Pull request merged**